s3.tf

```diff
@@ -74,6 +74,32 @@ resource "aws_s3_bucket" "apiary_data_bucket" {
   }
 }
 
+resource "aws_s3_bucket_versioning" "apiary_data_bucket_versioning" {
+  for_each = {
+    for schema in local.schemas_info : "${schema["schema_name"]}" => schema
+  }
+  bucket = each.value["data_bucket"]
+  versioning_configuration {
+    status = lookup(each.value, "s3_versioning_enabled", "Disabled")
+  }
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "apiary_data_bucket_versioning_lifecycle" {
+  for_each = {
+    for schema in local.schemas_info : "${schema["schema_name"]}" => schema
+  }
+  bucket = each.value["data_bucket"]
+  # Rule enabled when expiration max days is set
+  rule {
+    id     = "expire-noncurrent-versions-days"
+    status = lookup(each.value, "s3_versioning_expiration_days", "") != "" ? "Enabled" : "Disabled"
+
+    noncurrent_version_expiration {
+      noncurrent_days = tonumber(lookup(each.value, "s3_versioning_expiration_days", var.s3_versioning_expiration_days))
```

Comment:
Question, should we only configure the version expiration? Or should we also configure the max version retain , like 1,2,3 ? If only expiration configured, all older version will be deleted after the days meet.

Reply:
Replied privately, conclusion was to start only with days.